frontend/assets/javascripts/src/modules/analytics/facebook.js

```diff
@@ -1,48 +1,81 @@
-define(function() {
+/* global fbq, _fbq */
+/* eslint-disable no-unused-vars, no-underscore-dangle */
+define(['lodash/collection/forEach'], function(forEach) {
     'use strict';
 
-    function usSupporterLandingPage() {
+    var FACEBOOK_PROPERTY_ID = '869473446445427';
+
+    var URL_MAPPINGS = {
+        '/join/supporter/thankyou': recordConversion
+    };
+
+    var EVENT_MAPPINGS = {
+        '/': 'Lead',
+        '/choose-tier': 'Lead',
+
+        '/join/friend/enter-details': 'InitiateCheckout',
+        '/join/supporter/enter-details': 'InitiateCheckout',
+        '/join/partner/enter-details': 'InitiateCheckout',
+        '/join/patron/enter-details': 'InitiateCheckout',
+
+        '/join/friend/thankyou': 'CompleteRegistration',
+        '/join/supporter/thankyou': 'CompleteRegistration',
+        '/join/partner/thankyou': 'CompleteRegistration',
+        '/join/patron/thankyou': 'CompleteRegistration'
+    };
+
+    function callByPathname(mappings, callback) {
+        var pathname = window.location.pathname;
+        forEach(mappings, function(val, url) {
+            if(url === pathname && callback) {
```

Comment:
Perhaps the callback check can be done one, outside the loop. This may lead to a shorter code that uses `indexOf` on `EVENT_MAPPINGS` to determine if the current path is eligible.